### PR TITLE
[BugFix] Fixed incorrect schema id in CreateReplicaTask

### PR DESCRIPTION
**Pull request by sduzh**
CreateReplicaTask always uses the index id as the schema id, but for tables that have performed fast schema evolution, the schema id is not equal to the index id.

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by sduzh** [comment]
@sevev @wyb PTAL

**Review by sevev — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/SchemaInfo.java [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+            tSchema.setIndexes(tIndexes);
+        }
+
+        if (bloomFilterColumnNames != null) {
+            tSchema.setBloom_filter_fpp(bloomFilterFpp);
+        }
+        return tSchema;
+    }
+
+    public static Builder builder() {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/catalog/SchemaInfo.java:163]: can name to `newBuilder()` referring to protobuf Builder, `builder` interface is a little close to `build()`.

File: fe/fe-core/src/main/java/com/starrocks/catalog/SchemaInfo.java [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+        return new Builder();
+    }
+
+    public static class Builder {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/catalog/SchemaInfo.java:167]: can this code be generated by protobuf? so just define a protobuf and generating the corresponding Builder code. The `TTabletSchema` class just needs to implement a protobuf to `TTabletSchema` convert method.

Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/catalog/SchemaInfo.java:167]: The jprotobuf generated java code will not use the builder pattern.

File: fe/fe-core/src/main/java/com/starrocks/catalog/SchemaInfo.java (hunk 1 above)
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/catalog/SchemaInfo.java]: Updated

File: fe/fe-core/src/main/java/com/starrocks/catalog/SchemaInfo.java (hunk 2 above)
Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/catalog/SchemaInfo.java:167]: not necessary to be jprotobuf, just regular protobuf for auto code generation.

Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/catalog/SchemaInfo.java:167]: It seems that protobufs defined in StarRocks are compiled with jprotobuf by default, I may not be quite sure of the way you mean. Also, this class will be used in the PR I'm submitting later, and it needs to support being serialized by gson (so its members are marked with SeralizedName). java classes generated from protobuf can also be serialized by gson, but it needs to be done with an additional byte array conversion, and maybe that will make things more complicated?

Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/catalog/SchemaInfo.java:167]: ok

**Review by kevincai — APPROVED** [review]

**Review by sevev — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Pull request merged**